networkit/cpp/distance/test/DistanceGTest.cpp

```diff
@@ -160,41 +178,31 @@ TEST_F(DistanceGTest, testBidirectionalBFS) {
         bfs.run();
         BidirectionalBFS bbfs(G, source, target, true);
         bbfs.run();
-        if (bfs.distance(target) < G.upperNodeIdBound())
+
+        if (bfs.distance(target) < G.upperNodeIdBound()) {
+            // Source reaches target
             EXPECT_EQ(bfs.distance(target), bbfs.getDistance());
-        else
-            EXPECT_EQ(std::numeric_limits<count>::max(), bbfs.getDistance());
-        auto path = bbfs.getPath();
-        EXPECT_EQ(path.size(), bbfs.getDistance() - 1);
-        if (path.size()) {
-            EXPECT_TRUE(G.hasEdge(source, path.front()));
-            EXPECT_TRUE(G.hasEdge(path.back(), target));
-            for (count i = 1; i < path.size()-1; ++i)
-                EXPECT_TRUE(G.hasEdge(path[i], path[i + 1]));
+            const auto path = bbfs.getPath();
+            EXPECT_EQ(path.size(), bbfs.getDistance() - 1);
+
+            if (!path.empty()) { // At least two edges in the path
+                EXPECT_TRUE(G.hasEdge(source, path.front()));
+                EXPECT_TRUE(G.hasEdge(path.back(), target));
+                for (count i = 1; i < path.size()-1; ++i)
+                    EXPECT_TRUE(G.hasEdge(path[i], path[i + 1]));
+            } else
+                EXPECT_TRUE(G.hasEdge(source, target));
         } else
-            EXPECT_TRUE(G.hasEdge(source, target));
+            EXPECT_EQ(std::numeric_limits<edgeweight>::max(), bbfs.getDistance());
     };
-    for (int i = 0; i < 10; ++i) {
-        testGraph(G);
-        testGraph(G1);
-    }
 }
 
-TEST_F(DistanceGTest, testBidirectionalDijkstra) {
+TEST_P(DistanceGTest, testBidirectionalDijkstra) {
     Aux::Random::setSeed(42, false);
-    count n = 300;
-    Graph G(n, true, false);
-    Graph G1(n, true, true);
-    Graph gen = ErdosRenyiGenerator(n, 0.05, false).generate();
-    Graph gen1 = ErdosRenyiGenerator(n, 0.05, true).generate();
-    gen.forEdges([&](node u, node v) {
-        G.setWeight(u, v, Aux::Random::probability());
-    });
-    gen1.forEdges([&](node u, node v) {
-        G1.setWeight(u, v, Aux::Random::probability());
-    });
+    auto G = generateERGraph(500, isDirected() ? 0.05 : 0.02);
+    const auto infDist = std::numeric_limits<edgeweight>::max();
```

Comment:
Only minor: the value of `infDist` is used for several tests. Maybe it is a good idea to change it to be accessible by the complete class.